gpcontrib/gp_check_functions/gp_check_functions--1.3.sql

```diff
@@ -0,0 +1,706 @@
+                            'gp_move_orphaned_files_by_tablespace_location(tablespace_oid, /path) or ' ||
+                            'gp_move_orphaned_files_by_tablespace_location(ARRAY[''{tablespace_oid, /path}'', ...]) ' ||
+                            'to move files for specific tablespaces.';
+        END;
+END;
+$$;
+
+GRANT EXECUTE ON FUNCTION __gp_move_orphaned_files(
+    __gp_move_orphaned_files_pairs[], bool) TO public;
+
+--------------------------------------------------------------------------------
+-- @function:
+--        gp_move_orphaned_files
+--
+-- @in:
+--        target_location text
+--                Absolute path of the directory to which all orphaned files
+--                should be moved. The directory must be located on the same
+--                filesystem as each orphaned file on the segment; otherwise
+--                rename will fail.
+--
+-- @out:
+--        gp_segment_id int - segment content ID
+--        move_success bool - TRUE if the move succeeded
+--        oldpath text - absolute pathname before the move
+--        newpath text - absolute pathname after the move
+--
+-- @doc:
+--        Moves all orphaned files detected on a segment
+--        to a single target directory. If tablespaces reside on
+--        multiple filesystems, use
+--        gp_move_orphaned_files_by_tablespace_location instead.
+--
+--------------------------------------------------------------------------------
+-- NOTE: lock pg_class in shared mode and ensure no one transactions are running
+CREATE OR REPLACE FUNCTION gp_move_orphaned_files(target_location text) RETURNS TABLE (
+    gp_segment_id int,
+    move_success bool,
+    oldpath text,
+    newpath text
+)
+LANGUAGE SQL AS $$
+SELECT * FROM __gp_move_orphaned_files(
+    ARRAY[ROW(0, target_location)::__gp_move_orphaned_files_pairs],
+    process_all_tablespaces := TRUE
+);
+$$;
+
+GRANT EXECUTE ON FUNCTION gp_move_orphaned_files(text) TO public;
+
+--------------------------------------------------------------------------------
+-- @function:
+--        gp_move_orphaned_files_by_tablespace_location
+--
+-- @in:
+--        tablespace_oid oid
+--                OID of the tablespace whose orphaned files should be moved.
+--        target_location text
+--                Absolute path of the directory that will receive the files.
+--                Must be on the same filesystem as the tablespace location.
+--
+-- @out:
+--        gp_segment_id int - segment content ID
+--        move_success bool - TRUE if the move succeeded
+--        oldpath text - absolute pathname before the move
+--        newpath text - absolute pathname after the move
+--
+-- @doc:
+--        Moves orphaned files belonging to a specific tablespace
+--        to the supplied target directory.
+--
+--------------------------------------------------------------------------------
+-- NOTE: lock pg_class in shared mode and ensure no one transactions are running
```

Comment:
Does this note relate to the `gp_move_orphaned_files_by_tablespace_location` function? If yes, why not to move it under the `@doc` section? And in this case it is better to re-phase a little:
```suggestion
-- NOTE: locks pg_class in shared mode and ensures no transactions are running
```

The same comment is for the overloaded variant.

Reply:
Fixed